R/calc_variables_tree_level.R

```diff
@@ -0,0 +1,110 @@
+#' data_stems <- compose_stem_data(data_dendro, data_shoots)
+#' height_model <- load_height_models("C:/bosreservaten/Hoogtemodellen/")
+#' data_stems_calc <- calc_variables_stem_level(data_stems, height_model)
+#' calc_variables_tree_level(data_dendro, data_stems_calc)
+#' }
+#'
+#' @export
+#'
+#' @importFrom readr read_csv2
+#' @importFrom rlang .data
+#' @importFrom dplyr %>% bind_rows filter group_by left_join mutate n select summarise ungroup
+#'
+calc_variables_tree_level <-
+  function(data_dendro, data_stems_calc) {
+
+  data_dendro1 <- data_dendro %>%
+    select(
+      -.data$dbh_mm, -.data$tree_number, -.data$calc_height_fm,
+      -.data$intact_snag, -.data$decaystage
+    ) %>%
+    left_join(
+      data_stems_calc %>%
+        group_by(.data$plot_id, .data$tree_measure_id, .data$period) %>%
+        summarise(
+          tree_number = n(),
+          decaystage =
+            round(
+              sum(.data$decaystage * .data$dbh_mm ^ 2 / 4) /
+                sum(.data$dbh_mm ^ 2 / 4)
+            ),
+          intact_snag = max(.data$intact_snag),
+          dh_model = sum(.data$dh_model) >= 1,
+          calc_height_m = sum(.data$calc_height_m * .data$dbh_mm ^ 2 / 4) /
+            sum(.data$dbh_mm ^ 2 / 4),
+          calc_height_fm = sum(.data$calc_height_fm * .data$dbh_mm ^ 2 / 4) /
+            sum(.data$dbh_mm ^ 2 / 4),
+          calc_height_r = sum(.data$calc_height_r * .data$dbh_mm ^ 2 / 4) /
+            sum(.data$dbh_mm ^ 2 / 4),
+          dbh_mm = round(sqrt(sum(.data$dbh_mm ^ 2) / n())),
+          basal_area_m2 = sum(.data$basal_area_m2),
+          vol_bole_t1_m3 = sum(.data$vol_bole_t1_m3),
+          vol_bole_t2_m3 = sum(.data$vol_bole_t2_m3),
+          vol_bole_m3 = sum(.data$vol_bole_m3),
+          vol_crown_m3 = sum(.data$vol_crown_m3),
+          vol_tot_m3 = sum(.data$vol_tot_m3),
+          # RESULTS PER HECTARE
+          basal_area_alive_m2_ha = sum(.data$basal_area_alive_m2_ha),
+          basal_area_dead_m2_ha = sum(.data$basal_area_dead_m2_ha),
+          vol_alive_m3_ha = sum(.data$vol_alive_m3_ha),
+          vol_dead_standing_m3_ha = sum(.data$vol_dead_standing_m3_ha),
+          vol_bole_alive_m3_ha = sum(.data$vol_bole_alive_m3_ha),
+          vol_bole_dead_m3_ha = sum(.data$vol_bole_dead_m3_ha)
+        ) %>%
+        ungroup(),
+      by = c("plot_id", "tree_measure_id", "period")
+    ) %>%
+    mutate(
+      individual = (.data$ind_sht_cop == 10 | .data$ind_sht_cop == 12)
+    ) %>%
+    mutate(number_of_trees_alive_ha =
+             ifelse(
+               .data$alive_dead == 11,
+               .data$individual / .data$plotarea_ha,
+               0
+             ),
+           number_of_trees_dead_ha =
+             ifelse(
+               .data$alive_dead == 12,
+               .data$individual / .data$plotarea_ha,
+               0
+             )
+    ) %>%
+    select(-calc_height_fm, -calc_height_r, -dh_model, -vol_bole_t1_m3, -vol_bole_t2_m3, -individual)
```

Comment:
```suggestion
    select(-.data$calc_height_fm, -.data$calc_height_r, -.data$dh_model, -.data$vol_bole_t1_m3, -.data$vol_bole_t2_m3, -.data$individual)
```
Nog wat `.data`'s...  (Mogelijk moet dit not over 2 lijnen verdeeld worden)